File: life-assisitant-ui-desktop/src/api/notification.ts
import request from './request'
import type { Notification, UnreadCountResp, PageResult } from './types'

/**
 * 通知 API
 * 后端已落库（notification 表），全部为真实接口：
 * - 列表：GET /notifications?page=&page_size=（created_at 倒序）
 * - 未读数：GET /notifications/unread-count
 * - 全部已读：POST /notifications/mark-read
 * - 单条已读：POST /notifications/:id/read
 */
export const notificationApi = {
  /** 通知列表（分页，倒序） */
  list(params?: { page?: number; page_size?: number }) {
    return request.get<unknown, PageResult<Notification>>('/notifications', { params })
  },

  /** 获取未读通知数量 */
  unreadCount() {
    return request.get<unknown, UnreadCountResp>('/notifications/unread-count')
  },

  /** 标记所有通知已读 */
  markRead() {
    return request.post<unknown, { affected: number }>('/notifications/mark-read')
  },

  /** 标记单条通知已读 */
  markReadById(id: string) {
    return request.post<unknown, { affected: number }>(`/notifications/${id}/read`)
  },
}

File: life-assisitant-api/internal/controller/notification.go
package controller

import (
	"github.com/gogf/gf/v2/net/ghttp"

	"github.com/life-assistant/api/internal/consts/ecode"
	"github.com/life-assistant/api/internal/model/dto"
	"github.com/life-assistant/api/internal/response"
	"github.com/life-assistant/api/internal/service"
)

// NotificationController 通知控制器
type NotificationController struct{}

var Notification = &NotificationController{}

// List GET /api/v1/notifications
func (c *NotificationController) List(r *ghttp.Request) {
	var req dto.ListNotificationsReq
	if err := r.Parse(&req); err != nil {
		response.Error(r, ecode.ValidationFailed, err)
		return
	}
	out, err := service.Notification().List(r.Context(), &req)
	if err != nil {
		writeError(r, err)
		return
	}
	page := req.Page
	if page <= 0 {
		page = 1
	}
	pageSize := req.PageSize
	if pageSize <= 0 {
		pageSize = 20
	}
	response.Page(r, out.Items, out.Total, page, pageSize)
}

// UnreadCount GET /api/v1/notifications/unread-count
func (c *NotificationController) UnreadCount(r *ghttp.Request) {
	uid := r.GetCtxVar("user_id").String()
	if uid == "" {
		response.Error(r, ecode.AuthTokenMissing)
		return
	}
	out, err := service.Notification().UnreadCount(r.Context())
	if err != nil {
		writeError(r, err)
		return
	}
	response.Success(r, out)
}

// MarkRead POST /api/v1/notifications/mark-read
func (c *NotificationController) MarkRead(r *ghttp.Request) {
	uid := r.GetCtxVar("user_id").String()
	if uid == "" {
		response.Error(r, ecode.AuthTokenMissing)
		return
	}
	out, err := service.Notification().MarkRead(r.Context())
	if err != nil {
		writeError(r, err)
		return
	}
	response.Success(r, out)
}

// MarkReadByID POST /api/v1/notifications/:id/read
func (c *NotificationController) MarkReadByID(r *ghttp.Request) {
	id := r.Get("id").String()
	if id == "" {
		response.Error(r, ecode.MissingRequiredField)
		return
	}
	if err := service.Notification().MarkReadByID(r.Context(), id); err != nil {
		writeError(r, err)
		return
	}
	response.Success(r, map[string]any{"affected": 1})
}
